packages/compiler-cli/ngcc/src/packages/entry_point.ts

```diff
@@ -191,6 +192,7 @@ export function getEntryPointFormat(
       return 'esm2015';
     case 'esm5':
       return 'esm5';
+    case 'browser':
     case 'main':
       const mainFile = entryPoint.packageJson['main'];
       if (mainFile === undefined) {
```

Comment:
We need to guard for the case where it isn't a string, otherwise we may crash for packages that do have a complex browser property that we don't support.

```suggestion
      if (typeof mainFile !== 'string') {
```

Reply:
Thanks for picking this up.
It also made me look at the code again and realise an even more glaring mistake: when the property is `browser` we should not be doing `entryPoint.packageJson['main'];` 😱